### DISCUSSION TITLE
Implement standard macOS menu bar commands

### DISCUSSION THREAD

**Discussion by valentinegb** [godotengine/godot-proposals#7756] [Platforms]
On a Mac, it is expected that every application has the File, Edit, View, and Window command menus. Godot, however, does not have any of these! Some of the commands normally in these command menus are implemented in other Godot menus like Undo and Redo, but new users have to re-learn where there are if they don't use the keyboard shortcuts, and some commands just aren't implemented, like all of the Window commands. I believe it would make Godot feel much more familiar to macOS users if it implemented these standard command menus. I made a SwiftUI application to mockup what this might look like:

https://github.com/godotengine/godot-proposals/assets/35977727/179fa68a-fc3d-44e2-8f2b-b98707a1a038

A few notes regarding my mockup:

- I rearranged several commands to standard command menus. However, I realize that while it might be more intuitive for new users, it would probably throw off existing users.
- The icons in the Help menu are SF Symbols, which is what pretty much all of macOS uses for its symbols and icons. It *would* make Godot fit in a little better with macOS, but it honestly isn't a necessary change. The icons Godot currently uses there are a bit blurry / low resolution looking though, the SF Symbols icons look much clearer and cleaner.
- Some command names were slightly altered to match commands of stock macOS apps, for example by removing ellipsis on sub menus or making the two export commands clearly include context in their labels.

I made some small complimentary changes in my mockup, but really all I'm asking for is to add the standard command menus.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/7700.


Unfortunately, this would require making the menu ordering specific to macOS. Also, attempts to reorganize the editor toolbar have been poorly received by some users in the past: https://github.com/godotengine/godot/pull/60083#issuecomment-1207228133

**Reply by valentinegb** [reply]
That's totally fair, I did think that it might not be a good idea to reorder existing commands, but I think that at least adding the stock View and Window command menus would be okay and not disrupt existing users' experience.
